opentelemetry-auto-instrumentation/lib/opentelemetry-auto-instrumentation.rb

```diff
@@ -0,0 +1,159 @@
+      'ethon' => 'OpenTelemetry::Instrumentation::Ethon',
+      'excon' => 'OpenTelemetry::Instrumentation::Excon',
+      'faraday' => 'OpenTelemetry::Instrumentation::Faraday',
+      'grape' => 'OpenTelemetry::Instrumentation::Grape',
+      'graphql' => 'OpenTelemetry::Instrumentation::GraphQL',
+      'http_client' => 'OpenTelemetry::Instrumentation::HttpClient',
+      'mongo' => 'OpenTelemetry::Instrumentation::Mongo',
+      'mysql2' => 'OpenTelemetry::Instrumentation::Mysql2',
+      'net_http' => 'OpenTelemetry::Instrumentation::Net::HTTP',
+      'pg' => 'OpenTelemetry::Instrumentation::PG',
+      'que' => 'OpenTelemetry::Instrumentation::Que',
+      'racecar' => 'OpenTelemetry::Instrumentation::Racecar',
+      'rack' => 'OpenTelemetry::Instrumentation::Rack',
+      'rails' => 'OpenTelemetry::Instrumentation::Rails',
+      'rake' => 'OpenTelemetry::Instrumentation::Rake',
+      'rdkafka' => 'OpenTelemetry::Instrumentation::Rdkafka',
+      'redis' => 'OpenTelemetry::Instrumentation::Redis',
+      'restclient' => 'OpenTelemetry::Instrumentation::RestClient',
+      'resque' => 'OpenTelemetry::Instrumentation::Resque',
+      'ruby_kafka' => 'OpenTelemetry::Instrumentation::RubyKafka',
+      'sidekiq' => 'OpenTelemetry::Instrumentation::Sidekiq',
+      'sinatra' => 'OpenTelemetry::Instrumentation::Sinatra'
+    }.freeze
+
+    def self.detect_resource_from_env
+      env = ENV['OTEL_RUBY_RESOURCE_DETECTORS'].to_s
+      additional_resource = ::OpenTelemetry::SDK::Resources::Resource.create({})
+
+      env.split(',').each do |detector|
+        case detector
+        when 'container'
+          additional_resource = additional_resource.merge(::OpenTelemetry::Resource::Detector::Container.detect) if defined? ::OpenTelemetry::Resource::Detector::Container
+        when 'google_cloud_platform'
+          additional_resource = additional_resource.merge(::OpenTelemetry::Resource::Detector::GoogleCloudPlatform.detect) if defined? ::OpenTelemetry::Resource::Detector::GoogleCloudPlatform
+        when 'azure'
+          additional_resource = additional_resource.merge(::OpenTelemetry::Resource::Detector::Azure.detect) if defined? ::OpenTelemetry::Resource::Detector::Azure
+        when 'aws'
+          additional_resource = additional_resource.merge(::OpenTelemetry::Resource::Detector::AWS.detect(%i[ec2 ecs eks lambda])) if defined? ::OpenTelemetry::Resource::Detector::AWS
+        end
+      end
+
+      additional_resource
+    end
+
+    def self.determine_enabled_instrumentation
+      env = ENV['OTEL_RUBY_ENABLED_INSTRUMENTATIONS'].to_s
+
+      env.split(',').map { |instrumentation| OTEL_INSTRUMENTATION_MAP[instrumentation] }
+    end
+
+    def self.require_otel
+      return if @initialized
+
+      @initialized = true
+
+      lib = File.expand_path('..', __dir__)
+      $LOAD_PATH.reject! { |path| path.include?('opentelemetry-auto-instrumentation') }
+      $LOAD_PATH.unshift(lib)
```

Comment:
ok so this is simulating what bundler does to load gems. 

Do you also want to adding to `Gem.loaded_specs` or anything else that bundler would do?